client/app/pages/queries/utils/useQueryExecute.js

```diff
@@ -0,0 +1,33 @@
+import { useState, useMemo } from "react";
+import { includes } from "lodash";
+import useQueryResult from "@/lib/hooks/useQueryResult";
+import { useCallback } from "react";
+import { $location } from "@/services/ng";
+import { useEffect } from "react";
+
+function getMaxAge() {
+  const maxAge = $location.search().maxAge;
+  return maxAge !== undefined ? maxAge : -1;
+}
+
+export default function useQueryExecute(query) {
+  const [queryResult, setQueryResult] = useState(query.getQueryResult(getMaxAge()));
+  const queryResultData = useQueryResult(queryResult);
+  const queryExecuting = useMemo(() => !includes(["done", "failed"], queryResultData.status), [queryResultData.status]);
+
+  const executeQuery = useCallback(() => setQueryResult(query.getQueryResult(0)), [query]);
+
+  const executeQueryWithText = useCallback(
+    selectedQueryText => setQueryResult(query.getQueryResultByText(0, selectedQueryText)),
+    [query]
+  );
+
+  useEffect(() => {
+    if (!queryExecuting && queryResult.query_result.query === query.query) {
+      query.latest_query_data_id = queryResult.getId();
+      query.queryResult = queryResult;
+    }
+  }, [queryExecuting]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Not 100% sure why this logic exists, but it seems to keep the Query `latest_query_data` updated. (without having to GET the query again).

As for the `eslint-disable-line` I didn't want it to be triggering unnecessarily -- an alternative to such cases is creating a `usePrevious` hook.

Reply:
> it seems to keep the Query `latest_query_data` updated.

Yes, exactly.